content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,176 @@
+-->
+
+<!-- TODO Tab that shows SQL -->
+
+```prisma file=schema.prisma highlight=4;normal
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+Some databases, such as MongoDB or [PlanetScale](/guides/database/using-prisma-with-planetscale#differences-to-consider)),
+do not support foreign keys. For this situation, Prisma offers an alternative _relation mode_ where most properties of relations in relational databases are emulated.
+
+When using Prisma Client the behavior of queries is identical or similar, but all referential actions and most referential integrity is actually handled by the Prisma engine under the hood.
+
+<Admonition type="warning">
+  There are significant performance implications to emulating referential
+  integrity in the Prisma client. In cases where the underlying database can
+  handle referential integrity with foreign keys, it is usually the preferred
+  choice.
+</Admonition>
+
+</TopBlock>
+
+## How to set the relation mode in your Prisma schema
+
+To set the relation mode, you need to enable the `referentialIntegrity` preview feature. Add the `referentialIntegrity` feature flag to the `previewFeatures` list in the [`generator`](/reference/api-reference/prisma-schema-reference#generator) block of your Prisma schema file, and add the `relationMode` field to the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block:
+
+```prisma file=schema.prisma highlight=4,9;add
+datasource db {
+  provider     = "mysql"
+  url          = env("DATABASE_URL")
+  relationMode = "prisma"
+}
+
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialIntegrity"]
+}
+```
+
+To change the relation mode, update the `relationMode` field in the `datasource` block.
+
+<Admonition type="warning">
+
+The <inlinecode>relationMode</inlinecode> field was renamed in Prisma version 4.5.0,
+and was previously named <inlinecode>referentialIntegrity</inlinecode>.
+
+</Admonition>
+
+The available options are:
+
+- `foreignKeys`: this handles relations in the database using foreign keys. This is the default option when using all relational database connectors.
+- `prisma`: this emulates relations in the Prisma client. This is the default (and only available) option when using the MongoDB connector. You should also [enable this option](/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) when using the MySQL connector with a PlanetScale database.
```

Comment:
We could split this:

1. For relational databases: existing text (minus sentence about MongoDB)
2. MongoDB: no actual option, always set to `prisma` and emulation is active.

Reply:
How about:

> - `prisma`: this emulates relations and foreign key constraints in the Prisma client.
You should enable this option when using a database service that does not support foreign keys, like PlanetScale's MySQL connector.
Moreover, `prisma` is the default (and only available) option when using MongoDB as a database `provider`.

Reply:
Done